crates/ark/src/treesitter.rs

```diff
@@ -571,3 +571,22 @@ pub(crate) fn node_find_containing_call<'tree>(node: Node<'tree>) -> Option<Node
 
     None
 }
+
+pub(crate) fn point_end_of_previous_row(
+    mut point: tree_sitter::Point,
+    contents: &ropey::Rope,
+) -> tree_sitter::Point {
+    if point.row > 0 {
+        let prev_row = point.row - 1;
+        let line = contents.line(prev_row as usize);
+        let line_len = line.len_chars().saturating_sub(1); // Subtract 1 for newline
+        tree_sitter::Point {
+            row: prev_row,
+            column: line_len,
+        }
```

Comment:
Is there a reason you don't include the newline? I feel like the newline is technically just the end of the current line, right? So it feels like that position would be valid

Reply:
Because after the newline you are no longer on that line, so that position actually represents column 0 of the line we started with.

Side note, I find it confusing that Rope's line iterator includes trailing newlines. I think a "split by line" viewpoint is more intuitive.